### PR TITLE
web_log Virtual host enhancement and http/https

### PR DESCRIPTION
**Pull request by jgrossiord**
As I wanted to monitor a webserver hosting multiple websites, I wanted to have a quick view on which vhost has heavy load.
In the same time, I wanted to get a view on part of requests made in http and https.
So I added 2 graphs vhost and port giving the number of req/s per vhost and per port.

This can also be an answer to #2654 regarding vhost graphing.

Hope this helps, thank you for this wunderful product

### CODE REVIEW

**Comment by ktsaou** [comment]
Nice! Thank you!
Let's wait a bit more for @l2isbad to return from vacations.

Btw, please sign the CLA by editing CONTRIBUTORS.md and appending your name there.

File: python.d/web_log.chart.py [hunk 1]
```diff
@@ -707,6 +720,23 @@ def get_data_from_request_field(self, match_dict):
...
+                                                           'incremental'])
+                self.data[match_dict['port']] = 0
+            self.data[match_dict['port']] += 1
+        # requests per vhost
+        if match_dict.get('vhost'):
+            dim_id = match_dict['vhost'].replace('.', '_')
+            if dim_id not in self.data:
+                self.charts['vhost'].add_dimension([dim_id,
+                                                           match_dict['vhost'],
+                                                           'incremental'])
```
Comment (l2isbad) [inline python.d/web_log.chart.py]: fix the indentation please

File: python.d/web_log.chart.py [hunk 2]
```diff
@@ -128,6 +128,18 @@
                     'web_log.requests_per_user_defined', 'stacked'],
         'lines': [
             ['user_pattern_other', 'other', 'incremental', 1, 1]
+        ]},
+    'port': {
+        'options': [None, 'Requests Per Port', 'requests/s', 'port',
+                    'web_log.port', 'stacked'],
+        'lines': [
+            ['80', 'http', 'incremental', 1, 1],
+            ['443', 'https', 'incremental', 1, 1]
```
Comment (l2isbad) [inline python.d/web_log.chart.py]: add `port_` prefix to the dimension ids

File: python.d/web_log.chart.py [hunk 3]
```diff
@@ -328,7 +340,8 @@ def __init__(self, service):
                      'resp_time_upstream_avg': 0, 'unique_cur_ipv4': 0, 'unique_cur_ipv6': 0, '2xx': 0,
                      '5xx': 0, '3xx': 0, '4xx': 0, '1xx': 0, '0xx': 0, 'unmatched': 0, 'req_ipv4': 0,
                      'req_ipv6': 0, 'unique_tot_ipv4': 0, 'unique_tot_ipv6': 0, 'successful_requests': 0,
-                     'redirects': 0, 'bad_requests': 0, 'server_errors': 0, 'other_requests': 0, 'GET': 0}
+                     'redirects': 0, 'bad_requests': 0, 'server_errors': 0, 'other_requests': 0, 'GET': 0,
+                     'port_80': 0, 'port_443': 0}
```
Comment (l2isbad) [inline python.d/web_log.chart.py]: Why do we need this from start? It is added an empy chart (`'port_80': 0, 'port_443': 0` always 0) to all plugin users.
I suggest to not add zeroed `port_` keys here.

Comment (jgrossiord) [reply python.d/web_log.chart.py]: OK, I remove this

**Comment by ktsaou** [comment]
@l2isbad merge?

**Comment by l2isbad** [comment]
yes

**Pull request merged**